titan-compiler/parser.lua

```diff
@@ -214,11 +226,11 @@ local grammar = pg.compile([[
                      / (COLON NAME funcargs)            -> suffix_methodcall
                      / (LBRACKET exp RBRACKET)          -> suffix_index
 
-    simpleexp       <- (NIL -> tonil)                   -> Exp_Value
-                     / (FALSE -> tofalse)               -> Exp_Value
-                     / (TRUE -> totrue)                 -> Exp_Value
-                     / (NUMBER)                         -> Exp_Value
-                     / (STRING)                         -> Exp_Value
+    simpleexp       <- (NIL)                            -> nil_exp
+                     / (FALSE -> tofalse)               -> Exp_Bool
+                     / (TRUE -> totrue)                 -> Exp_Bool
```

Comment:
We may want to split the tag Bool into True and False to comply with our manual:
> The type number is syntax sugar for integer | float, and boolean is syntax sugar for true | false.

Reply:
Good point but maybe we should leave union types for v1...